xbmc/platform/freebsd/network/NetworkFreebsd.cpp

```diff
@@ -92,23 +91,18 @@ bool CNetworkInterfaceFreebsd::GetHostMacAddress(unsigned long host_ip, std::str
   struct rt_msghdr* rtm;
   struct sockaddr_inarp* sin;
   struct sockaddr_dl* sdl;
-  int mib[6];
+  constexpr std::array<int, 6> mib = {
+      CTL_NET, PF_ROUTE, 0, AF_INET, NET_RT_FLAGS, RTF_LLINFO,
+  };
 
   mac = "";
 
-  mib[0] = CTL_NET;
-  mib[1] = PF_ROUTE;
-  mib[2] = 0;
-  mib[3] = AF_INET;
-  mib[4] = NET_RT_FLAGS;
-  mib[5] = RTF_LLINFO;
-
-  if (sysctl(mib, ARRAY_SIZE(mib), NULL, &needed, NULL, 0) == 0)
+  if (sysctl(mib.data(), mib.size(), NULL, &needed, NULL, 0) == 0)
```

Comment:
why not `const_cast<int*>(mib.data())` like in [NetworkOsx.cpp](https://github.com/xbmc/xbmc/blob/33fd69de20d89eabf98857048da031cfc510f879/xbmc/platform/darwin/osx/network/NetworkOsx.cpp#L72) ?

Reply:
I guess the freebsd implementation allows a `const` type.